### PR TITLE
Jenkins file

### PR DESCRIPTION
**Pull request by fsufitch**

### CODE REVIEW

File: JenkinsFile [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            [$class: 'UsernamePasswordMultiBinding', credentialsId: "${env.PCF_CREDS}", usernameVariable: "CFUSER", passwordVariable: "CFPASS"]
+          ]) {
+              sh """
+                cf api ${env.PCF_API_ENDPOINT}
+                cf auth ${CFUSER} ${CFPASS}
+                cf target -o ${env.PCF_ORG} -s ${pcfSpace}
+                cf push ${fullAppName} -f manifest.jenkins.yml --hostname ${fullAppName} -d ${pcfDomain} --no-start
+                cf set-env ${fullAppName} SPACE ${pcfSpace}
+                cf set-env ${fullAppName} DOMAIN ${pcfDomain}
+                cf start ${fullAppName}
```
Comment (go0ty) [inline JenkinsFile]: https://github.com/venicegeo/pz-gateway/blob/master/JenkinsFile#L108

File: JenkinsFile [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            }
+        }*/
+    }
+
+    def appvers = sh(script: """git describe --long --tags --always | sed 's/\\./-/'g""", returnStdout: true)
+    appvers = appvers.trim()
+    def fullAppName = "${appName}-${appvers}"
+
+    stage("Archive") {
+        def archiveName = "${appName}.tgz"
```
Comment (go0ty) [inline JenkinsFile:73]: Did you check Nexus to make sure that it placed the artifact in the proper place?

File: JenkinsFile [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+    stage("Phase One Deploy") {
+      deployPhase(env.PHASE_ONE_PCF_SPACE, env.PHASE_ONE_PCF_DOMAIN)
+    }
+    if(!params.SKIP_INTEGRATION_TESTS) {
+        stage("Phase One Integration Tests") {
+          integrationTestPhase(env.PHASE_ONE_PCF_SPACE, env.PHASE_ONE_PCF_DOMAIN)
+        }
+    }
+    stage("Phase Two Deploy") {
```
Comment (go0ty) [inline JenkinsFile]: Make a check like this https://github.com/venicegeo/pz-gateway/blob/master/JenkinsFile#L168

Before doing the phase 2 deploy.

File: JenkinsFile [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+      deployPhase(env.PHASE_TWO_PCF_SPACE, env.PHASE_TWO_PCF_DOMAIN)
+    }
+    if(!params.SKIP_INTEGRATION_TESTS) {
```
Comment (go0ty) [inline JenkinsFile]: Also use the same check https://github.com/venicegeo/pz-gateway/blob/master/JenkinsFile#L168 before doing the phase 2 integration tests

File: JenkinsFile [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+      withCredentials([
+        file(credentialsId: "${POSTMAN_SECRET_FILE}", variable: "POSTMAN_FILE")
+      ]) {
+          withEnv([
+            "PATH+=${root}/integration-tests/node_modules/newman/bin:${nodejs}/bin",
+            "PCF_SPACE=${pcfSpace}",
+            "HOME=${WORKSPACE}"
+          ]) {
+              dir ("integration-tests") {
+                  sh "ci/blackbox.sh"
```
Comment (go0ty) [inline JenkinsFile]: I don't see a ci/blackbox.sh in https://github.com/venicegeo/bftest-integration/tree/master/ci

Comment (go0ty) [reply JenkinsFile]: You can copy and paste from here

https://github.com/venicegeo/bf-tideprediction/blob/master/JenkinsFile#L147

File: JenkinsFile (hunk 2 above)
Comment (fsufitch) [reply JenkinsFile:73]: Yep. Though the only purpose this serves is really just to not re-deploy the same version.

**Pull request merged**